src/mongocxx/include/mongocxx/v1/bulk_write.hpp

```diff
@@ -33,105 +56,1014 @@ class bulk_write {
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL() ~options();
+
+    ///
+    /// Move constructor.
+    ///
+    /// @par Postconditions:
+    /// - `other` is in an assign-or-destroy-only state.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL() options(options&& other) noexcept;
+
+    ///
+    /// Move assignment.
+    ///
+    /// @par Postconditions:
+    /// - `other` is in an assign-or-destroy-only state.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(options&) operator=(options&& other) noexcept;
+
+    ///
+    /// Copy construction.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL() options(options const& other);
+
+    ///
+    /// Copy assignment.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(options&) operator=(options const& other);
+
+    ///
+    /// Default initialization.
+    ///
+    /// @par Postconditions:
+    /// - All supported fields are "unset" or zero-initialized.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL() options();
+
+    ///
+    /// Set the "bypassDocumentValidation" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(options&) bypass_document_validation(bool bypass_document_validation);
+
+    ///
+    /// Return the current "bypassDocumentValidation" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(bsoncxx::v1::stdx::optional<bool> const) bypass_document_validation() const;
+
+    ///
+    /// Set the "comment" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(options&) comment(bsoncxx::v1::types::value comment);
+
+    ///
+    /// Return the current "comment" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(bsoncxx::v1::stdx::optional<bsoncxx::v1::types::view> const) comment() const;
+
+    ///
+    /// Set the "let" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(options&) let(bsoncxx::v1::document::value let);
+
+    ///
+    /// Return the current "let" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(bsoncxx::v1::stdx::optional<bsoncxx::v1::document::view> const) let() const;
+
+    ///
+    /// Set the "ordered" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(options&) ordered(bool ordered);
+
+    ///
+    /// Return the current "ordered" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(bool) ordered() const;
+
+    ///
+    /// Set the "writeConcern" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(options&) write_concern(v1::write_concern wc);
+
+    ///
+    /// Return the current "writeConcern" field.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(bsoncxx::v1::stdx::optional<v1::write_concern>) write_concern() const;
+};
+
+///
+/// The result of a write operation.
+///
+/// Supported fields include:
+/// - `inserted_count` ("nInserted" or "insertedCount")
+/// - `matched_count` ("nMatched" or "matchedCount")
+/// - `modified_count` ("nModified" or "modifiedCount")
+/// - `deleted_count` ("nDeleted" or "deletedCount")
+/// - `upserted_count` ("nUpserted" or "upsertedCount")
+/// - `upserted_ids` ("upserted" or "upsertedIds")
+///
+/// @important The raw server response is translated by mongoc into the Bulk Write API specification format even when
```

Comment:
Remove documentation? I expect the `reply` document returned by `mongoc_bulk_operation_execute` is not exposed directly to the user.